src/devices/wifi/nm-device-iwd.c

```diff
@@ -809,6 +814,27 @@ check_scanning_prohibited (NMDeviceIwd *self, gboolean periodic)
 	return prohibited;
 }
 
+static void
+scan_cb (GObject *source, GAsyncResult *res, gpointer user_data)
+{
+	NMDeviceIwd *self = user_data;
+	NMDeviceIwdPrivate *priv = NM_DEVICE_IWD_GET_PRIVATE (self);
```

Comment:
Below, we pass `priv->cancellable` to `g_dbus_proxy_call()`. That is good, usually, either an async request must be cancellable, or we must take an additional reference to keep the object alive. Taking an additional reference is usually bad, because it means the object cannot be destroyed, until the callback returns. So, cancel is good.

But, in the callback you must *first* check whether 
```
    if (g_error_matches (error, G_IO_ERROR, G_IO_ERROR_CANCELLED))
```
before accessing `self`.

Reply:
Ok, good point.  I'm new to GCancellables.